Marginalia appears to break with Clojure 1.7.0

I'm guessing because of use of LispReader internals, though I haven't dug that deep yet. Possibly see https://github.com/Engelberg/instaparse/pull/94 http://dev.clojure.org/jira/browse/CLJ-1424

```$ lein with-profile doc docs
Unpacking resources...
Destination directory exists, skipping unpack. lein clean, if necessary
Generating Marginalia documentation for the following source files:
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/data_migrations.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/db.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/dev_server.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/etags.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/migrate.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/resources.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/service.clj
   /Users/michael.blume/workspace/seed-advisor/src/com/climate/seed_advisor/utils.clj

Exception in thread "main" java.lang.RuntimeException: Problem parsing near line 1 <require> original reported cause is clojure.lang.ArityException: Wrong number of args (4) passed to: parser/read-keyword -- clojure.lang.ArityException: Wrong number of args (4) passed to: parser/read-keyword, compiling:(/private/var/folders/dl/2qwtfptd4wb3wsxp3htbp9ddpc_zl9/T/form-init6498617338864557319.clj:1:125)
	at clojure.lang.Compiler.load(Compiler.java:7249)
	at clojure.lang.Compiler.loadFile(Compiler.java:7175)
	at clojure.main$load_script.invoke(main.clj:275)
	at clojure.main$init_opt.invoke(main.clj:280)
	at clojure.main$initialize.invoke(main.clj:308)
	at clojure.main$null_opt.invoke(main.clj:343)
	at clojure.main$main.doInvoke(main.clj:421)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Problem parsing near line 1 <require> original reported cause is clojure.lang.ArityException: Wrong number of args (4) passed to: parser/read-keyword -- clojure.lang.ArityException: Wrong number of args (4) passed to: parser/read-keyword
	at clojure.lang.LispReader.read(LispReader.java:284)
	at clojure.lang.LispReader.read(LispReader.java:195)
	at clojure.lang.LispReader.read(LispReader.java:190)
	at marginalia.parser$parse_STAR_$fn__418$fn__421.invoke(parser.clj:153)
	at marginalia.parser$parse_STAR_$fn__418.invoke(parser.clj:152)
	at clojure.core$repeatedly$fn__5108.invoke(core.clj:4919)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:507)
	at clojure.core$seq__4125.invoke(core.clj:135)
	at clojure.core$tree_seq$walk__5042$fn__5043.invoke(core.clj:4737)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.LazySeq.more(LazySeq.java:85)
	at clojure.lang.RT.more(RT.java:683)
	at clojure.core$rest__4111.invoke(core.clj:73)
	at clojure.core$flatten.invoke(core.clj:6847)
	at marginalia.parser$parse_STAR_.invoke(parser.clj:146)
	at marginalia.parser$parse.invoke(parser.clj:382)
	at marginalia.parser$parse_file.invoke(parser.clj:400)
	at marginalia.core$path_to_doc.invoke(core.clj:177)
	at clojure.core$map$fn__4550.invoke(core.clj:2622)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:507)
	at clojure.core$seq__4125.invoke(core.clj:135)
	at clojure.core$map$fn__4550.invoke(core.clj:2614)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:507)
	at clojure.core$seq__4125.invoke(core.clj:135)
	at clojure.core$map$fn__4550.invoke(core.clj:2614)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:507)
	at clojure.core$seq__4125.invoke(core.clj:135)
	at clojure.core$apply.invoke(core.clj:628)
	at marginalia.hiccup$eval57$fn__58.invoke(hiccup.clj:99)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at clojure.lang.Var.invoke(Var.java:379)
	at marginalia.html$toc_html.invoke(html.clj:189)
	at marginalia.html$uberdoc_html.invoke(html.clj:400)
	at marginalia.core$uberdoc_BANG_.invoke(core.clj:208)
	at marginalia.core$run_marginalia.doInvoke(core.clj:293)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at user$eval869.invoke(form-init6498617338864557319.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6792)
	at clojure.lang.Compiler.eval(Compiler.java:6782)
	at clojure.lang.Compiler.load(Compiler.java:7237)
	... 11 more
Error encountered performing task 'docs' with profile(s): 'doc'
Subprocess failed```

--- a/src/marginalia/parser.clj
+++ b/src/marginalia/parser.clj
@@ -74,24 +74,27 @@
       (directive))
     (not directive)))
 
-(defn read-comment [reader semicolon]
-  (let [sb (StringBuilder.)]
-    (.append sb semicolon)
-    (loop [c (.read reader)]
-      (let [ch (char c)]
-        (if (or (= ch \newline)
-                (= ch \return))
-          (let [line (dec (.getLineNumber reader))
-                text (.toString sb)
-                include? (process-directive! text)]
-            (when (and include? (comments-enabled?))
-              (swap! *comments* conj {:form (Comment. text)
-                                      :start line
-                                      :end line}))
-            reader)
-          (do
-            (.append sb (Character/toString ch))
-            (recur (.read reader))))))))
+(defn read-comment
+  ([reader semicolon]
+   (let [sb (StringBuilder.)]
+     (.append sb semicolon)
+     (loop [c (.read reader)]
+       (let [ch (char c)]
+         (if (or (= ch \newline)
+                 (= ch \return))
+           (let [line (dec (.getLineNumber reader))
+                 text (.toString sb)
+                 include? (process-directive! text)]
+             (when (and include? (comments-enabled?))
+               (swap! *comments* conj {:form (Comment. text)
+                                       :start line
+                                       :end line}))
+             reader)
+           (do
+             (.append sb (Character/toString ch))
+             (recur (.read reader))))))))
+  ([reader semicolon opts pending]
+   (read-comment reader semicolon)))
 
 (defn set-comment-reader [reader]
   (aset (get-field clojure.lang.LispReader :macros nil)
@@ -116,15 +119,18 @@
                [String]
                nil s))
 
-(defn read-keyword [reader colon]
-  (let [c (.read reader)]
-    (if (= (int \:) c)
-      (-> (read-token reader (char c))
-          match-symbol
-          DoubleColonKeyword.)
-      (do (.unread reader c)
-          (-> (read-token reader colon)
-              match-symbol)))))
+(defn read-keyword
+  ([reader colon]
+   (let [c (.read reader)]
+     (if (= (int \:) c)
+       (-> (read-token reader (char c))
+           match-symbol
+           DoubleColonKeyword.)
+       (do (.unread reader c)
+           (-> (read-token reader colon)
+               match-symbol)))))
+  ([reader colon opts pending]
+   (read-keyword reader colon)))
 
 (defn set-keyword-reader [reader]
   (aset (get-field clojure.lang.LispReader :macros nil)

